integration/spark/src/main/common/java/io/openlineage/spark/agent/lifecycle/BaseVisitorFactory.java

```diff
@@ -0,0 +1,74 @@
+package io.openlineage.spark.agent.lifecycle;
+
+import io.openlineage.client.OpenLineage;
+import io.openlineage.spark.agent.lifecycle.plan.AppendDataVisitor;
+import io.openlineage.spark.agent.lifecycle.plan.BigQueryNodeVisitor;
+import io.openlineage.spark.agent.lifecycle.plan.CommandPlanVisitor;
+import io.openlineage.spark.agent.lifecycle.plan.InsertIntoDataSourceDirVisitor;
+import io.openlineage.spark.agent.lifecycle.plan.InsertIntoDataSourceVisitor;
+import io.openlineage.spark.agent.lifecycle.plan.InsertIntoDirVisitor;
+import io.openlineage.spark.agent.lifecycle.plan.InsertIntoHadoopFsRelationVisitor;
+import io.openlineage.spark.agent.lifecycle.plan.InsertIntoHiveDirVisitor;
+import io.openlineage.spark.agent.lifecycle.plan.InsertIntoHiveTableVisitor;
+import io.openlineage.spark.agent.lifecycle.plan.LogicalRDDVisitor;
+import io.openlineage.spark.agent.lifecycle.plan.LogicalRelationVisitor;
+import io.openlineage.spark.agent.lifecycle.plan.QueryPlanVisitor;
+import io.openlineage.spark.agent.lifecycle.plan.SaveIntoDataSourceCommandVisitor;
+import io.openlineage.spark.agent.lifecycle.plan.VisitorFactory;
+import io.openlineage.spark.agent.lifecycle.plan.wrapper.InputDatasetVisitor;
+import io.openlineage.spark.agent.lifecycle.plan.wrapper.OutputDatasetVisitor;
+import io.openlineage.spark.agent.lifecycle.plan.wrapper.OutputDatasetWithMetadataVisitor;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.spark.sql.SQLContext;
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan;
+
+public abstract class BaseVisitorFactory implements VisitorFactory {
+
+  protected List<QueryPlanVisitor<? extends LogicalPlan, OpenLineage.Dataset>>
+      getBaseCommonVisitors(SQLContext sqlContext, String jobNamespace) {
+    List<QueryPlanVisitor<? extends LogicalPlan, OpenLineage.Dataset>> list = new ArrayList<>();
+    list.add(new LogicalRelationVisitor(sqlContext.sparkContext(), jobNamespace));
+    list.add(new LogicalRDDVisitor());
+    list.add(new CommandPlanVisitor(new ArrayList<>(list)));
+    if (BigQueryNodeVisitor.hasBigQueryClasses()) {
+      list.add(new BigQueryNodeVisitor(sqlContext));
+    }
+    return list;
+  }
+
+  public abstract List<QueryPlanVisitor<? extends LogicalPlan, OpenLineage.Dataset>>
```

Comment:
This shouldn't be public- it should only be called internally, so make it `protected`